pandas/tests/series/methods/test_replace.py

```diff
@@ -387,6 +387,16 @@ def test_replace_categorical(self, categorical, numeric):
             expected = expected.cat.add_categories(2)
         tm.assert_series_equal(expected, result)
 
+    @pytest.mark.parametrize(
+        "data, data_exp", [(["a", "b", "c"], ["b", "b", "c"]), (["a"], ["b"])]
+    )
+    def test_replace_categorical_inplace(self, data, data_exp):
+        # GH 53358
+        result = pd.Series(data, dtype="category")
+        result.replace(to_replace="a", value="b", inplace=True)
+        expected = pd.Series(data_exp, dtype="category")
+        tm.assert_index_equal(expected.dtype.categories, result.dtype.categories)
```

Comment:
Is this not covered by tm.assert_series_equal?

Reply:
Thank you, I replaced `tm.assert_index_equal` with `tm.assert_series_equal`.